src/axom/multimat/multimat.hpp

```diff
@@ -0,0 +1,765 @@
+   * Entry at `idx` should containing 'true' where materials are presented in
+   * the cell.\n
+   * The number of materials and cell must be set prior to calling this function
+   * with setNumberOfMaterials(int) and setNumberOfCells(int)
+   *
+   * \param A boolean vector of size num_mats * num_cells containing information
+   * on if a materials is present in a cell.
+   *
+   */
+  void setCellMatRel(std::vector<bool>& relation_info, DataLayout layout);
+
+  //functions related to fields
+
+  int getNumberOfFields() const { return m_mapVec.size(); }
+
+  /**
+   * \brief Add a field to the MultiMat object
+   *
+   * \tparam T The data type (double, float...) of the field
+   * \param field_name The name of the field, used to retrieve the field later
+   * \param field_mapping
+   * \param data_layout
+   * \param sparsity_layout
+   * \param data_array The array containing data to the field. The length of the
+   *            array should be `num_mats * num_cells * ncomp` if the current
+   *            format is Dense, or `num_nonzero * ncomp` if the current format
+   *            is Sprase.
+   * \param (optional) ncomp The number of component of the field. Default is 1
+   * \return int the index of the field, can be used to retrieve the field later
+   */
+  template <class T>
+  int addField(const std::string& field_name,
+               FieldMapping field_mapping,
+               DataLayout data_layout,
+               SparsityLayout sparsity_layout,
+               T* data_array,
+               int ncomp = 1);
+
+private:
+  template <typename T>
+  int addFieldArray_impl(const std::string&,
+                         FieldMapping,
+                         DataLayout,
+                         SparsityLayout,
+                         T*,
+                         int);
+
+public:
+  /**
+   * \brief Set the volume fraction field
+   * \detail volume fraction field is assumed to be a double. Its field index is
+   * always 0, and the name of the field is "Volfrac"
+   *
+   * \param data_array the array containing the volumn fraction information
+   * \return int the volume fraction field index, which is always zero.
+   */
+  int setVolfracField(double* data_array,
+                      DataLayout layout,
+                      SparsityLayout sparsity);
+
+  /**
+   * \brief Search for and return the field index of the field given its name.
+   *
+   * \param field_name the name of the field
+   * \return int the index of the field
+   */
+  int getFieldIdx(const std::string& field_name) const;
+
+  /**
+   * \brief Search for and return the field given the field name.
+   * \detail the field is of type Field1D, containing an entry for each cell
+   * or material. To retrieve a field of type Field2D, use get2dField()
+   *
+   * \tparam T The data type of the field
+   * \param field_name the name of the field
+   * \return Field1D<T>& the field reference
+   */
+
```

Comment:
Move this blank line two lines down, to bring this declaration up next to its Doxygen and separate it from the next Doxygen comment.  (May sound silly, but it will help a lot with readability.)